src/player.js

```diff
@@ -1715,6 +1733,15 @@ export default class Player extends FakeEventTarget {
     };
   }
 
+  _maybeCreateAdsController(): void {
+    if (!this._adsController) {
+      const adsPluginController = this._controllerProvider.getAdsController();
+      if (adsPluginController) {
+        this._adsController = new AdsController(this, adsPluginController);
```

Comment:
doesn't seem to use both `player` and `adsPluginController`

Reply:
see my comment below